source/ShipModel.cpp

```diff
@@ -88,7 +88,30 @@ bool ShipModel::createAllTask(int data) {
 
 bool ShipModel::failAllTask() {
 	for (int i = 0; i < donuts.size(); i++) {
-		float angle = (float)(rand() % (int)(getSize()));
+		float angle = 0;
+		bool goodAngle = false;
+		while (!goodAngle) {
+			angle = (float)(rand() % (int)(getSize()));
+			for (unsigned int k = 0; k < breaches.size(); k++) {
+				float breachAngle = breaches[k]->getAngle();
+				float diff = shipSize / 2 - abs(abs(breachAngle - angle) - shipSize / 2);
+				if (breachAngle == -1 || diff > MIN_DISTANCE) {
+					goodAngle = true;
+				} else {
+					goodAngle = false;
+					continue;
```

Comment:
What is the point of this continue at the end of the loop?
Also doesn't the written value of `goodAngle` get overriden on the next iteration still?